subprojects/core/src/main/java/org/gradle/api/internal/file/copy/CopyFileVisitorImpl.java

```diff
@@ -65,7 +69,7 @@ private void processFile(FileVisitDetails visitDetails) {
     }
 
     private DefaultFileCopyDetails createDefaultFileCopyDetails(FileVisitDetails visitDetails) {
-        return instantiator.newInstance(DefaultFileCopyDetails.class, visitDetails, copySpecResolver, fileSystem);
+        return instantiator.newInstance(DefaultFileCopyDetails.class, visitDetails, copySpecResolver, objectFactory, fileSystem);
```

Comment:
I wonder if passing down the `instantiator` is not enough. From what I can see you only use the `newInstance` method.

Reply:
At first, I was trying to utilize the `instantiator` to provide an `ObjectFactory` instance. But this doesn't work because it only knows about `default service` so it doesn't have `ObjectFactory` in its scope and cannot inject it. I'm not sure if this behavior is intended but I can say that it is confusing.

I don't think I can create a `Property` instance without the `ObjectFactory` because there is some state in the latter.

Reply:
Indeed, I misunderstood that last usage. All good.